packages/core/src/UserState.ts

```diff
@@ -172,8 +182,12 @@ export default class UserState {
         return foundLeaf.index
     }
 
-    getEpochKeys(_epoch?: bigint | number, nonce?: number) {
-        const epoch = _epoch ?? this.sync.calcCurrentEpoch()
+    async getEpochKeys(
+        _epoch?: bigint | number,
+        nonce?: number,
+        attesterId: bigint | string = this.sync.attesterId
+    ) {
+        const epoch = _epoch ?? (await this.sync.calcCurrentEpoch(attesterId))
```

Comment:
Also very against `getEpochKeys` being async, this causes more frontend problems.

Reply:
[813fda7](https://github.com/Unirep/Unirep/pull/367/commits/813fda7b49c24bbc1cebcaf9e548b4f66d3095f8)